jest-circus is missing dependency on jest-runtime

## 🐛 Bug Report

After upgrading Jest to 25.1.0, I noticed an issue with `jest-circus` having a dev-only dependency on `jest-runtime`.

```js
  ● Test suite failed to run

    Cannot find module 'jest-runtime'

      at Object.<anonymous> (node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:5:43)
```

`jest-runtime` is indeed referenced within jestAdapter.ts here:

https://github.com/facebook/jest/blob/2ee214d47f8bbd7fec1b6b4051a2e265136c79a5/packages/jest-circus/src/legacy-code-todo-rewrite/jestAdapter.ts#L12

It's referenced as a dev dependency here:

https://github.com/facebook/jest/blob/2ee214d47f8bbd7fec1b6b4051a2e265136c79a5/packages/jest-circus/package.json#L38

## To Reproduce

I can create a reproduction repo if necessary, seems like this can be solved fairly easily by just moving the dependency though.

## Expected behavior

No uncaught exceptions from within the test framework.

## Link to repl or repo (highly encouraged)

See note above.

<!--
Please provide either a [repl.it demo](https://repl.it/languages/jest) or a minimal repository on GitHub.
Issues without a reproduction link are likely to stall.
-->

## envinfo

<!--
Run npx envinfo --preset jest
Paste the results here:
-->

```
System:
    OS: macOS 10.15.3
    CPU: (4) x64 Intel(R) Core(TM) i7-7567U CPU @ 3.50GHz
  Binaries:
    Node: 8.11.1 - /usr/local/bin/node
    Yarn: 1.17.3 - /usr/local/bin/yarn
    npm: 5.6.0 - /usr/local/bin/npm
  npmPackages:
    jest: 25.1.0 => 25.1.0
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,8 @@
 
 ### Fixes
 
+- `[jest-circus]` Fix type ellision of jest-runtime imports ([#9717](https://github.com/facebook/jest/pull/9717))
+
 ### Chore & Maintenance
 
 ### Performance

--- a/packages/jest-circus/package.json
+++ b/packages/jest-circus/package.json
@@ -28,6 +28,7 @@
     "jest-each": "^25.2.3",
     "jest-matcher-utils": "^25.2.3",
     "jest-message-util": "^25.2.3",
+    "jest-runtime": "^25.2.3",
     "jest-snapshot": "^25.2.3",
     "jest-util": "^25.2.3",
     "pretty-format": "^25.2.3",
@@ -41,8 +42,7 @@
     "@types/babel__traverse": "^7.0.4",
     "@types/co": "^4.6.0",
     "@types/stack-utils": "^1.0.1",
-    "execa": "^3.2.0",
-    "jest-runtime": "^25.2.3"
+    "execa": "^3.2.0"
   },
   "engines": {
     "node": ">= 8.3"

--- a/packages/jest-circus/src/legacy-code-todo-rewrite/jestAdapter.ts
+++ b/packages/jest-circus/src/legacy-code-todo-rewrite/jestAdapter.ts
@@ -9,7 +9,7 @@ import * as path from 'path';
 import type {Config} from '@jest/types';
 import type {JestEnvironment} from '@jest/environment';
 import type {TestResult} from '@jest/test-result';
-import Runtime = require('jest-runtime');
+import type {RuntimeType as Runtime} from 'jest-runtime';
 import type {SnapshotStateType} from 'jest-snapshot';
 
 const FRAMEWORK_INITIALIZER = require.resolve('./jestAdapterInit');

--- a/packages/jest-jasmine2/src/index.ts
+++ b/packages/jest-jasmine2/src/index.ts
@@ -10,7 +10,7 @@ import type {Config, Global} from '@jest/types';
 import type {AssertionResult, TestResult} from '@jest/test-result';
 import type {JestEnvironment} from '@jest/environment';
 import type {SnapshotStateType} from 'jest-snapshot';
-import Runtime = require('jest-runtime');
+import type {RuntimeType as Runtime} from 'jest-runtime';
 
 import {getCallsite} from '@jest/source-map';
 import installEach from './each';

--- a/packages/jest-runtime/src/index.ts
+++ b/packages/jest-runtime/src/index.ts
@@ -64,6 +64,8 @@ type CacheFS = {[path: string]: string};
 
 namespace Runtime {
   export type Context = JestContext;
+  // ditch this export when moving to esm - for now we need it for to avoid faulty type elision
+  export type RuntimeType = Runtime;
 }
 
 const testTimeoutSymbol = Symbol.for('TEST_TIMEOUT_SYMBOL');

